chips/esp32/src/timg.rs

```diff
@@ -194,9 +194,14 @@ impl<F: time::Frequency, const C3: bool> time::Time for TimG<'_, F, C3> {
     type Ticks = Ticks64;
 
     fn now(&self) -> Self::Ticks {
-        // a write (of any value) to T0UPDATE stores the
-        // current counter value to T0LO and T0HI
+        // According to the datasheet, a write (of any value) to T0UPDATE stores
+        // the current counter value to T0LO and T0HI. However, in my (bradjc)
+        // testing the first read of t0lo and t0hi after a single write to
+        // t0update returned the old counter value. Writing two values to
+        // t0update seems to fix this.
         self.registers.t0update.set(0xABC);
+        self.registers.t0update.set(0x8000_0000);
+
```

Comment:
<img width="845" height="264" alt="Image" src="https://github.com/user-attachments/assets/a0267f41-826b-49e1-bf95-1b213f9e5c37" />

I believe this needs to loop/poll until `t0update.get() == 0`